Norm_end_to_end.Rmd

```diff
@@ -432,200 +524,450 @@ colnames(study_summary) <- colnames(study_sample)
-#' Calculate flags 
-flagging <- vapply(model_lm,flag_model_inj_range, logical(1),
-                   min_range = diff(range(res$injection_index)) * req_prop)
+#' Calculate flags
+flagging <- vapply(qc_lm, flag_model_inj_range, logical(1),
+                   min_range = diff(range(res$injection_index)) * req_prop,
+                   column = "injection_index")
 
 ##' Plot the distribution of slopes.
-hist(slps_lm, breaks = 128, xlab = "slope", main = "lm")
-hist(slps_lm[which(flagging)], breaks = 128, add = TRUE,
+hist(qc_slps, breaks = 128, xlab = "slope", main = "lm")
+hist(qc_slps[which(flagging)], breaks = 128, add = TRUE,
      col = "#0000ff80")
 
 #' Split into excluded and good mdls
-mdls_excluded <- model_lm[which(flagging)]
+mdls_excluded <- qc_lm[which(flagging)]
 
 #' remove model fits for flagged slopes
-model_lm[unique(which(flagging))] <- NA
-slps_lm[unique(which(flagging))] <- NA
+qc_lm[unique(which(flagging))] <- NA
+qc_slps[unique(which(flagging))] <- NA
 
 fts_adj <- sort(unique(
-    names(model_lm)[!is.na(model_lm)]))
+    names(qc_lm)[!is.na(qc_lm)]))
 ```
 
+
+
 The table below lists the number of features for which the model was fitted and
 the number of features for which model fitting was skipped or discarded.
 
-```{r fit-model-table, message=FALSE, include=FALSE, results=}
-tab <- cbind(dataset = c(length(model_lm),
-                         length(which(flagging)),
-                         sum(!is.na(model_lm)))
-             )
+```{r fit-model-table, message=FALSE, include=TRUE, results="asis"}
+tab <- data.frame(dataset = c(length(qc_lm),
+                              length(which(flagging)),
+                              sum(!is.na(qc_lm)))
+                  )
 rownames(tab) <- c("total features", "low inj idx range",
                    "valid model fits")
 cptn <- paste("Numbers of features for which an injection index dependent",
               "model could be fitted.")
 pandoc.table(tab, style = "rmarkdown", caption = cptn)
 ```
 
-```{r features-large-slopes-plots, message=FALSE, warning=FALSE, include=FALSE}
+```{r features-large-slopes-plots, message=FALSE, warning=FALSE, include=TRUE}
 #' Plotting all features with an absolute slope larger than some value.
 #' Plots are created but not displayed here.
 #' Features with absolute slope > 0.025, maybe more ?
 dr <- "largest_slopes/above_below_0.025/"
-dir.create(dr, showWarnings = FALSE)
-fts <- names(model_lm)[which(abs(slps_lm) > 0.025)]
+dir.create(dr, showWarnings = FALSE, recursive = TRUE)
+fts <- names(qc_lm)[which(abs(qc_slps) > 0.025)]
 for (i in fts) {
     png(paste0(dr, i, ".png"), width = 14, height = 7, pointsize = 6,
         res = 200, units = "cm")
     plot(x = colData(res)$injection_index,
          y = log2(assay(res, "median_filled_500")[i, ]),
          xlim = range(colData(res)$injection_index, na.rm = TRUE),
-         ylim = range(log2(assay(res, "median_filled_500")[i, ]), na.rm = TRUE))
-    abline(0, slps_lm[i], col = "red")
```

Comment:
better to just plot the linear model - which can be done with `abline` and the fitted linear model (see below).